pkg/distributor/distributor.go

```diff
@@ -814,6 +815,44 @@ func (d *Distributor) prePushRelabelMiddleware(next PushFunc) PushFunc {
 				req.Timeseries[tsIdx].RemoveLabel(labelName)
 			}
 
+			if len(ts.Labels) == 0 {
+				removeTsIndexes = append(removeTsIndexes, tsIdx)
+				continue
+			}
+		}
+
+		if len(removeTsIndexes) > 0 {
```

Comment:
Does this need to be present in both middlewares?

Reply:
I kept it in `prePushSortAndFilterMiddleware` as a safety mechanism, and then it's also in the relabel middleware since relabeling can drop series: https://github.com/grafana/mimir/pull/7326/files#diff-8e51742acf8f88fa2ae01855fb19125feb5c54c41023f1eaa8beded9f1b4bcc3R806-R808